### PR TITLE
[Upgrade] `az upgrade`: Change MSI URL based on the installed CLI architecture

### PR DESCRIPTION
**Pull request by bebound**
**Description**

Since we've released 64-bit MSI, `az upgrade` also need to be updated to match the current architecture.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/util/custom.py [hunk 1]
```diff
@@ -185,8 +185,14 @@ def _upgrade_on_windows():
     Directly installing from URL may be blocked by policy: https://github.com/Azure/azure-cli/issues/19171
     This also gives the user a chance to manually install the MSI in case of msiexec.exe failure.
     """
-    logger.warning("Updating Azure CLI with MSI from https://aka.ms/installazurecliwindows")
-    tmp_dir, msi_path = _download_from_url('https://aka.ms/installazurecliwindows')
+    import platform
+
+    if platform.architecture()[0] == '32bit':
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/custom.py:190]: [`platform.architecture`](https://docs.python.org/3/library/platform.html#platform.architecture) returns the bit architecture of the Python executable:

```py
> C:\Users\xxx\AppData\Local\Programs\Python\Python310-32\python.exe -c "import platform; print(platform.architecture())"
('32bit', 'WindowsPE')
> C:\Users\xxx\AppData\Local\Programs\Python\Python310\python.exe -c "import platform; print(platform.architecture())"
('64bit', 'WindowsPE')
```

Do we want to detect the machine's bit architecture instead?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/util/custom.py:190]: [`platform.machine`](https://docs.python.org/3/library/platform.html#platform.machine) seems to be the candidate for detecting the machine's architecture:

```
> & $env:LOCALAPPDATA\Programs\Python\Python310-32\python.exe -c "import platform; print(platform.machine())"
AMD64
> & $env:LOCALAPPDATA\Programs\Python\Python310\python.exe -c "import platform; print(platform.machine())"
AMD64
```

Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/util/custom.py:190]: No, I guess most users are using a 64-bit OS. I don't want to force them use 64-bit and break their extensions unexpectedly.

File: src/azure-cli/azure/cli/command_modules/util/custom.py [hunk 2]
```diff
@@ -185,8 +185,14 @@ def _upgrade_on_windows():
...
+        msi_url = 'https://aka.ms/installazurecliwindows'
+    else:
+        msi_url = 'https://aka.ms/installazurecliwindowsx64'
```
[the remark this answers is not in the document]
Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/util/custom.py:193]: I want to save a `pylint: disable=line-too-long` comment.
It's also clearer to see that the difference in the URL is the `x64` suffix.

File: src/azure-cli/azure/cli/command_modules/util/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/util/custom.py:190]: Fair enough.

**Review by jiasli — APPROVED** [review]

**Pull request merged**